internal/worker/server.go

```diff
@@ -122,9 +121,9 @@ func (s *Server) WatchHeartbeats() {
 				logrus.Panicf("Cannot marshal the heartbeat error: %v", err)
 			}
 
-			err = s.FinishJob(token, resJson)
+			err = s.jobs.RequeueOrFinishJob(id, 2, resJson)
```

Comment:
s.FinishJob makes a call to prometheus.FinishJobMetrics ; by calling the queue directly here this will now be missed by heartbeats.

Reply:
My bad! d'oh! Will fix.